kafka-connect-s3/src/main/java/io/confluent/connect/s3/format/bytearray/ByteArrayRecordWriterProvider.java

```diff
@@ -56,15 +61,24 @@ public String getExtension() {
   @Override
   public RecordWriter getRecordWriter(final S3SinkConnectorConfig conf, final String filename) {
     return new RecordWriter() {
-      final S3OutputStream s3out = storage.create(filename, true);
+      final String adjustedFilename = getAdjustedFilename(filename, getExtension());
+      final S3OutputStream s3out = storage.create(adjustedFilename, true);
       final OutputStream s3outWrapper = s3out.wrapForCompression();
 
       @Override
       public void write(SinkRecord record) {
-        log.trace("Sink record: {}", record);
+        log.trace("Sink record with view {}: {}", recordView, record);
         try {
-          byte[] bytes = converter.fromConnectData(
-              record.topic(), record.valueSchema(), record.value());
+          byte[] bytes;
+          if (recordView instanceof HeaderRecordView) {
+            JsonConverter jsonConverter = new JsonConverter();
```

Comment:
we can't create a new converter instance with every record. 
We have to cache it.